addons/sourcemod/scripting/vsh/modifiers/modifiers_ice.sp

```diff
@@ -124,8 +119,5 @@ public Action Timer_ResetSpeed(Handle hTimer, int iRef)
 	if (iClient <= 0 || iClient > MaxClients || !IsClientInGame(iClient))
 		return;
 	
-	g_flPlayerSpeedMultiplier[iClient] /= ICE_SLOWDOWN;
-	
-	//Recalculate player's speed
-	TF2_AddCondition(iClient, TFCond_SpeedBuffAlly, 0.01);
+	TF2_StunPlayer(iClient, 0.0, 0.0, TF_STUNFLAG_SLOWDOWN);
```

Comment:
`Timer_ResetSpeed` callback can be deleted when there `TF2_StunPlayer` with `ICE_DURATION`? Otherwise removing `TFCond_Dazed` cond also works.